src/RoleDB.php

```diff
@@ -61,67 +62,12 @@ public static function createFromRootMetadata(RootMetadata $rootMetadata)
             }
             $roleInfo['paths'] = [];
 
-            $db->addRole($roleName, $roleInfo);
+            $db->roles[$roleName] = $roleInfo;
         }
 
         return $db;
     }
 
-    /**
-     * Constructs a new RoleDB object.
-     */
-    public function __construct()
-    {
-        $this->roles = [];
-    }
-
-    /**
-     * Adds role metadata to the database.
-     *
-     * @param string $roleName
-     *     The role name.
-     * @param array $roleInfo
-     *     An associative array of role metadata, including:
-     *     - keyids: An array of authorized public key signatures for the role.
-     *     - threshold: An integer for the threshold of signatures required for
-     *       this role.
-     *     - paths: An array of the path patterns that this role may sign.
-     *
-     * @return void
-     *
-     * @throws \Exception
-     *     Thrown if the role already exists.
-     *
-     * @see https://github.com/theupdateframework/specification/blob/master/tuf-spec.md#4-document-formats
-     *
-     * @todo Provide more complete documentation of the structure once
-     *     delgation is implemented and fixtures are regenerated. Issues:
-     *     - https://github.com/php-tuf/php-tuf/issues/50
-     *     - https://github.com/php-tuf/php-tuf/issues/52
-     */
-    public function addRole(string $roleName, array $roleInfo)
-    {
-        if ($this->roleExists($roleName)) {
-            throw new \Exception('Role already exists: ' . $roleName);
-        }
-
-        $this->roles[$roleName] = $roleInfo;
-    }
-
-    /**
-     * Verifies whether a given role name is stored in the role database.
-     *
-     * @param string $roleName
-     *     The role name.
-     *
-     * @return boolean
-     *     True if the role is found in the role database; false otherwise.
-     */
-    public function roleExists(string $roleName)
-    {
-        return !empty($this->roles[$roleName]);
-    }
-
```

Comment:
I'd rather remove this any other such dead code in a separate, dedicated issue, with code commends and followup issues to revisit it (similar to https://github.com/php-tuf/php-tuf/issues/21 ).

Reply:
Restored all this.